### PR TITLE
Add bundle and cluster state, jsonPointer example

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

File: docs/gitrepo-structure.md [hunk 1]
```diff
@@ -207,3 +207,40 @@ will target `deployment.yaml`.  The patch will be applied using JSON Merge, Stra
...
+
+**WaitApplied**: Bundles have been synced from controlplane and downstream cluster, but are waiting to be deployed.
+
+**ErrApplied**: Bundles have been synced from controlplane and downstream cluster, but there are some errors when deploying the bundle.
+
+**OutOfSync**: Bundles have been synced from controlplane, but downstream agent hasn't synced the change yet.
+
+**Pending**: Bundles are being processed by controlplane.
+
+**Modified**: Bundles have been deployed and all resources are ready, but there are some resources modified from outside changes.
```
Comment (catherineluse) [inline docs/gitrepo-structure.md:228]: What's the definition of outside changes? Any changes not made by the Fleet controller?

Can we say

> **Modified**: Bundles have been deployed and all resources are ready, but there are some changes that were not made by the Fleet controller.

File: docs/gitrepo-structure.md [hunk 2]
```diff
@@ -207,3 +207,40 @@ will target `deployment.yaml`.  The patch will be applied using JSON Merge, Stra
 Which strategy is used is based on the file content. Even though JSON strategies are used, the files can be written
 using YAML syntax.
  
+## Cluster and Bundle state
+
+Clusters and Bundles has different states in different phases when applying bundles.
```
Comment (catherineluse) [inline docs/gitrepo-structure.md:212]: > Clusters and Bundles have different states in each phase of applying Bundles.

File: docs/gitrepo-structure.md [hunk 3]
```diff
@@ -207,3 +207,40 @@ will target `deployment.yaml`.  The patch will be applied using JSON Merge, Stra
 Which strategy is used is based on the file content. Even though JSON strategies are used, the files can be written
 using YAML syntax.
  
+## Cluster and Bundle state
```
Comment (catherineluse) [inline docs/gitrepo-structure.md:210]: Cluster and Bundle State

File: docs/gitrepo-structure.md [hunk 4]
```diff
@@ -207,3 +207,40 @@ will target `deployment.yaml`.  The patch will be applied using JSON Merge, Stra
...
+
+Clusters and Bundles has different states in different phases when applying bundles.
+
+For Bundles:
+
+**Ready**: Bundles have been deployed and all resources are ready.
+
+**NotReady**: Bundles have been deployed and some resources are not ready.
+
+**WaitApplied**: Bundles have been synced from controlplane and downstream cluster, but are waiting to be deployed.
```
Comment (catherineluse) [inline docs/gitrepo-structure.md:220]: > Bundles have been synced from the Fleet controller and the downstream cluster, but are waiting to be deployed.

File: docs/gitrepo-structure.md [hunk 5]
```diff
@@ -207,3 +207,40 @@ will target `deployment.yaml`.  The patch will be applied using JSON Merge, Stra
...
+
+**ErrApplied**: Bundles have been synced from controlplane and downstream cluster, but there are some errors when deploying the bundle.
```
Comment (catherineluse) [inline docs/gitrepo-structure.md:222]: > Bundles have been synced from the Fleet controller and the downstream cluster,

**Review by catherineluse — CHANGES_REQUESTED** [review]
I think it's ok to say 'downstream cluster.' If you want to say it differently, another option could be a 'Fleet managed cluster.'

File: docs/gitrepo-structure.md (hunk 1 above)
Comment (StrongMonkey) [reply docs/gitrepo-structure.md:228]: I think it is better to say?
> Bundles have been deployed and all resources are ready, but there are some changes that were not made by git repository.

**Review by nickgerace — APPROVED** [review]
After @catherineluse's suggestions, LGTM

**Review by catherineluse — APPROVED** [review]
lgtm

**Pull request merged**